### PR TITLE
added reminder output option, and default dollar-quoting based on db client

### PR DESCRIPTION
**Pull request by zacronos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,69 +52,84 @@ dbsync will be used, or globally.
...
+                       improvement to resource usage by dbsync).  The default value for this
+                       option is true when --client is set to 'pg', and false otherwise.  This
+                       is not relevant when --migration-at-once is also used.  For more details
+                       on dollar-quoting, see section 4.1.2.4 (Dollar-quoted String Constants)
+                       of http://www.postgresql.org/docs/9.3/static/sql-syntax-lexical.html     
+--command-buffering    This sets the number of SQL commands to buffer before pausing reading
+                       from the file; must be a positive integer. This is a performance-tuning
+                       option and shouldn't need to be altered for most use cases.  [default: 4]
+--stack-traces         If set, stack traces will be logged with any errors (when present).
 ```
```
Comment (zacronos) [inline README.md:137]: made the option help a little narrower so it displays better on github

File: src/cli.coffee [hunk 2]
```diff
@@ -90,11 +90,20 @@ optionSpecs =
     describe: "If set, dbsync will not restrict the migrations performed to only those that have not run successfully
               before; this is useful for rerunning previously-run scripts during development, or for scripting misc db
               commands without requiring any additional client tools to be installed."
+  reminder:
+    requiresArg: true
+    describe: "Some environments where dbsync may run (e.g. CircleCI) require periodic output to ensure the process is
+              running properly.  Sometimes a migration could take a long time, but without this indicating a problem.
+              In such cases, you can use --reminder to request reminder output every X minutes (a best-effort attempt is
+              made, so to be safe you should set a lower value than you really need); a value of 0 disables reminder
+              outout.  Value must be non-negative, but can be a decimal (e.g. --reminder .5)"
```
Comment (zacronos) [inline src/cli.coffee:99]: added this "reminder" option

File: src/cli.coffee [hunk 3]
```diff
@@ -90,11 +90,20 @@ optionSpecs =
...
   "dollar-quoting":
     boolean: true
-    describe: "If set, dbsync will allow dollar-quoted strings as specified by PostgreSQL. This is not relevant when
-              --migration-at-once is also used.  For more details, see section 4.1.2.4. (Dollar-quoted String
-              Constants) of 
+    describe: "If set, this will force dbsync to allow dollar-quoted strings as specified by PostgreSQL. The negation,
+              --no-dollar-quoting, is also available to force quoting to be turned off (this may make a minor
+              improvement to resource usage by dbsync).  The default value for this option is true when --client is set
+              to 'pg', and false otherwise.  This is not relevant when --migration-at-once is also used.  For more
+              details on dollar-quoting, see section 4.1.2.4 (Dollar-quoted String Constants) of
```
Comment (zacronos) [inline src/cli.coffee:106]: changed this one so it uses an intelligent default

**Comment by zacronos** [comment]
@nmccready

File: src/migrator.coffee [hunk 4]
```diff
@@ -29,9 +33,12 @@ validateOptions = (optionsToCheck) ->
   if optionsToCheck["on-read-error"]? && ["ignore", "log", "exit"].indexOf(optionsToCheck["on-read-error"]) == -1
     throw "invalid value for 'on-read-error': '#{optionsToCheck["on-read-error"]}'"
 
-  if optionsToCheck["command-buffering"]? && !isPositiveInteger(optionsToCheck["command-buffering"])
+  if optionsToCheck["command-buffering"]? && !validateNumber(optionsToCheck["command-buffering"], min: 1, integer: true)
```
Comment (nmccready) [inline src/migrator.coffee]: may want to be explicit here with `{min: 1, integer: true}` or bring it to a new line

```
!validateNumber(optionsToCheck["command-buffering"], 
  min: 1
  integer: true)
```

File: src/cli.coffee (hunk 2 above)
Comment (nmccready) [inline src/cli.coffee:99]: cool

**Comment by nmccready** [comment]
LGTM, do u want to skip any other mod suggestions?

**Comment by zacronos** [comment]
@nmccready, I've pushed up your suggestion, if you have others feel free to fire away

**Pull request merged**